packages/cli/src/commands/BaseCommand.ts

```diff
@@ -115,11 +136,9 @@ export abstract class BaseCommand extends Command {
 		await this.externalHooks.init();
 	}
 
-	async initLicense(instanceType: N8nInstanceType = 'main'): Promise<void> {
-		config.set('generic.instanceType', instanceType);
-
+	async initLicense(): Promise<void> {
 		const license = Container.get(License);
-		await license.init(this.instanceId, instanceType);
+		await license.init(this.instanceId, this.instanceType ?? 'main');
```

Comment:
This was why I wanted to create a `ServerCommand` class, so that all clie commands don't end up using `instanceType='main'`. But that can be fixed in another PR.